From e4e91d36f860678c9608bdbf3aa2188ff651c169 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 24 Sep 2013 17:41:53 +0300
Subject: [PORT FROM MAIN] Telephony: Fix critical KW errors

BZ: 85540

Fixes null pointer access critical KW errors
in CatService and GsmServiceStateTracker.

Change-Id: I96748737354c4ad0c8c4d24f93eaf3de739448f5
Orig-Change-Id: I82c624223e0fa7d0a34cd2f27bec67024daeeed0
Category: device enablement
Domain: Telephony-Modem-Common
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/cat/CatService.java |    2 +-
 .../telephony/gsm/GsmServiceStateTracker.java      |    6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/java/com/android/internal/telephony/cat/CatService.java b/src/java/com/android/internal/telephony/cat/CatService.java
index 7ad7015..271a112 100644
--- a/src/java/com/android/internal/telephony/cat/CatService.java
+++ b/src/java/com/android/internal/telephony/cat/CatService.java
@@ -801,7 +801,7 @@ public class CatService extends Handler implements AppInterface {
     }
 
     private boolean validateResponse(CatResponseMessage resMsg) {
-        if (mCurrntCmd != null) {
+        if (mCurrntCmd != null && resMsg != null) {
             return (resMsg.mCmdDet.compareTo(mCurrntCmd.mCmdDet));
         }
         return false;
diff --git a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
index 8f300e4..2f510e5 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
@@ -1364,8 +1364,10 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
         boolean equalsMcc = true;
 
         try {
-            equalsMcc = simNumeric.substring(0, 3).
-                    equals(operatorNumeric.substring(0, 3));
+            if (operatorNumeric != null) {
+                equalsMcc = simNumeric.substring(0, 3).
+                        equals(operatorNumeric.substring(0, 3));
+            }
         } catch (Exception e){
         }
         return equalsMcc;
-- 
1.7.4.1

